Source/PINDiskCache.m

```diff
@@ -1389,20 +1421,52 @@ - (void)setWritingProtectionOption:(NSDataWritingOptions)writingProtectionOption
     NSDataWritingOptions option = NSDataWritingFileProtectionMask & writingProtectionOption;
     
     [strongSelf lock];
-    strongSelf->_writingProtectionOption = option;
+        strongSelf->_writingProtectionOption = option;
     [strongSelf unlock];
   } withPriority:PINOperationQueuePriorityHigh];
 }
 #endif
 
+- (void)lockUntilWritable
+{
+    __unused int result = pthread_mutex_lock(&_mutex);
+    NSAssert(result == 0, @"Failed to lock PINDiskCache %@. Code: %d", self, result);
+    
+    // spinlock if the disk isn't writable
+    while (_diskWritable == NO) {
+        [self unlock];
+        usleep(100);
+        
+        __unused int result = pthread_mutex_lock(&_mutex);
```

Comment:
Why not call [self lock] here and in the method below?

Reply:
Good point.